src/csv/csv.service.ts

```diff
@@ -12,13 +26,15 @@ export class CsvService {
       const stream = Readable.from(fileBuffer);
 
       stream
-        .pipe(csv())
-        .on('data', (data) => results.push(data as T))
+        .pipe(csv({ strict: true }))
+        .on('data', (data) => {
+          results.push(data as T);
+        })
         .on('end', () => {
           resolve(results);
         })
         .on('error', (error) => {
-          reject(error);
+          reject(new BadRequestException(`${error}`));
```

Comment:
not sure if this service is responsible of throwing that kind of error : 
IMO : services throws `InvalidInputData` + some details about line in file (maybe `error` already carries this information)
controller catches that exception to throw back a `BadRequestException`

Reply:
The pris error, code and message are sometimes shady and say Unknow error. Wanted to wrap it to be clearer.